docs(mTLS): Clarifies the general client config for mTLS (#7637)
strimzi/strimzi-kafka-operator: documentation/modules/security/con-securing-client-authentication.adoc

= User authentication

[role="_abstract"]

> Use the `KafkaUser` custom resource to configure an authentication mechanism for clients that require access to a Kafka cluster. 

Reviewer: I think this is a bit confusing. `e an authentication mechanism for clients` would be configured in the `Kafka` CR in the `.spec.kafka.listeners` section. This configures what kind of credentials will be issued for the user => TLS certificate, SCRAM-SHA password etc..

Author: Changed to:

> Use the `KafkaUser` custom resource to configure authentication credentials for users (clients) that require access to a Kafka cluster. 
> Configure the credentials using the `authentication` property in `KafkaUser.spec`.

Configure user authentication using the `authentication` property in `KafkaUser.spec`.

> Enable an authentication mechanism using the `type` field.

Reviewer: Same as above, this does not enable the mechanism. It just controls the type of credentials which will be generated.

  # ...
----

> When configuring clients outside the Kubernetes cluster, the authentication type must match the equivalent configuration for the `Kafka` listener used to access the Kafka cluster.

Reviewer: This applied in general. Not just outside Kubernetes cluster.

  # ...
----

> The authentication type must match the equivalent configuration for the `Kafka` listener used to access the Kafka cluster.

Reviewer: The authorization is cluster wide, not per-listener.

```suggestion
The authentication type must match the equivalent configuration in the `Kafka` custom resource.
```

Author: @scholzj This is authentication here, not authorization. 
That's per-listener, right?

I've added a line to _User authorization_ though:

> Authorization is cluster-wide.
> The authorization type must match the equivalent configuration in the `Kafka` custom resource.

  user.password: # Password for protecting the PKCS #12 archive file
----

When you configure a client, you specify the following:

> * *Truststore* properties for the public cluster CA certificate to authenticate with the Kafka cluster

Reviewer: Maybe this would be more clear to regular users

```suggestion
* *Truststore* properties for the public cluster CA certificate to verify the identity of the Kafka cluster
```

* *Keystore* properties for the user authentication credentials to verify the client

The configuration depends on the file format (PEM or PKCS #12).

> This example uses the PKCS #12 archive files, and the passwords required to access the credentials. 

Reviewer: You call it `archive` on multiple places, but I do not think it is right. It is `store` or just `file`. I think calling it `archive` suggests it is some kind of ZIP for example, but it is more than that.

Reviewer 2: Agree, let's use `store` it's more appropriate.
